### DISCUSSION TITLE
Complex order by

### DISCUSSION THREAD

**Discussion by bsor-dev** [prisma/prisma#19575] [Q&A] [answered]
I have a problem with ordering of character varying type, its alphanumeric data.

I get a wrong order for desc

41
418
42

I should get

418
42
41

This query works but there's no order by raw option in prisma? How can I achieve this without using raw query
```sql
SELECT id,
       name,
       TYPE,
       city,
       country,
       "orgid",
       "postalcode",
       status,
       street,
       "createdat"
FROM   public."company"
ORDER  BY ( Substring("orgid", '^[0-9]+') ) :: INT DESC,
          Coalesce(Substring("orgid", '[^0-9_].*$'), '') DESC 
```

**Comment by ludralph** [comment, marked as answer]
Hi @bsor-dev 👋 

Unfortunately, it is not possible to express this in a native Prisma Client query at the moment. However, we do have a feature request to support this behavior as seen in this issue. 

- https://github.com/prisma/prisma/issues/6574

It would be great if you could share your use case on the issue and leave a 👍 in the issue to help us prioritize.

Since you need to express this using raw SQL, you can check out https://safeql.dev/compatibility/prisma.html as it helps you write better SQL with Prisma.

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by nurul3101** [reply]
@bsor-dev I'll close this discussion as resolved. Please let us know in case you still have any queries.
